src/backend/utils/adt/numeric.c

```diff
@@ -5934,10 +5937,21 @@ bigint_poly_sum(PG_FUNCTION_ARGS)
 					(errcode(ERRCODE_NUMERIC_VALUE_OUT_OF_RANGE),
 						errmsg("Arithmetic overflow error converting expression to data type bigint.")));
 		}
-		else
-			PG_RETURN_INT64((int64) result);
+		else {
+			if (isSum)
+				PG_RETURN_INT64((int64) result);
+			else
```

Comment:
add a comment to make clear that else here means avg only

Reply:
Fixed in the latest commit